engine/baml-runtime/src/runtime/runtime_interface.rs

```diff
@@ -76,17 +86,34 @@ impl<'a> InternalClientLookup<'a> for InternalBamlRuntime {
                 #[cfg(not(target_arch = "wasm32"))]
                 let clients = &self.clients;
 
-                if let Some(client) = clients.get(client_name) {
-                    Ok(client.clone())
-                } else {
-                    let walker = self
-                        .ir()
-                        .find_client(client_name)
-                        .context(format!("Could not find client with name: {}", client_name))?;
-                    let client = LLMProvider::try_from((&walker, ctx)).map(Arc::new)?;
-                    clients.insert(client_name.into(), client.clone());
-                    Ok(client)
+                // if a client exists, check if the env vars have changed
+                if clients.contains_key(client_name) {
+                    // make sure to clone the client to avoid holding a lock, otherwise dashmap will deadlock!
+                    let client = clients.get(client_name).map(|c| c.clone()).unwrap();
+                    // if the env vars haven't changed, return the cached client
+                    if !client.has_env_vars_changed(ctx.env_vars()) {
+                        return Ok(client.provider.clone());
+                    }else{
```

Comment:
Could you enable format on save, sometimes we run into conflict issues because of formatting